xcell/mappers/mapper_P18CMBK.py

```diff
@@ -94,17 +94,25 @@ def get_nl_coupled(self):
     def get_cl_fiducial(self):
         if self.cl_fid is None:
             ell = self.get_ell()
-            cl = self.noise[2] - self.noise[1]
+            noise = self._get_noise()
+            cl = noise[2] - noise[1]
             cl = interp1d(self.noise[0], cl, bounds_error=False,
                           fill_value=(cl[0], cl[-1]))(ell)
             self.cl_fid = np.array([cl])
         return self.cl_fid
 
+    def _get_noise(self):
+        if self.noise is None:
+            # Read noise file. Column order is: ['l', 'Nl', 'Nl+Cl']
+            self.noise = np.loadtxt(self.config['file_noise'], unpack=True)
+
+        return self.noise
+
     def get_ell(self):
         return np.arange(3 * self.nside)
 
     def get_ells_in_table(self):
```

Comment:
do we ever use this function?

Reply:
Yes, in https://github.com/xC-ell/xCell/blob/80919022f167d010579a1266e4eb7d523a1091ca/xcell/cls/to_sacc.py#L152. Maybe a bit redundant, though.

Reply:
wait, but that's `get_ell`, not `get_ells_in_table`, right?

Reply:
oh, sorry. I thought you meant `get_ell`. Yes, `get_ells_in_table` is not use anywhere. Removed.